### PR TITLE
macro based MonitoredAwait

### PR DESCRIPTION
**Pull request by ymatsuda**
- My first scala macro :)
- Created new module "macros" to compile the macro definition before anything else
- MonitoredAwait gets the class/method name of the caller at compile time. It no longer inspects the stack trace. This should be faster and generate less garbages.
- Added the class/method name to airbrake messages
- Timing log is disabled. do monitoredAwait.logging.result(...) if you want to log the timing.

@eishay , @andrewconner, please review. Thanks!

### CODE REVIEW

**Comment by andrewconner** [comment]
Nice!

**Comment by eishay** [comment]
cool!

File: kifi-backend/project/Build.scala [hunk 1]
```diff
@@ -159,10 +159,14 @@ object ApplicationBuild extends Build {
       Keys.fork := false
     )
 
-    lazy val common = play.Project("common", appVersion, commonDependencies, path = file("modules/common")).settings(
+    lazy val macros = play.Project("macros", appVersion, commonDependencies, path = file("modules/macros")).settings(
```
Comment (andrewconner) [inline kifi-backend/project/Build.scala:162]: Does this need to be a `play.Project`? I think we can use normal sbt `Project` (http://www.playframework.com/documentation/2.1.x/SBTSubProjects) for non-Play projects.

Comment (ymatsuda) [inline kifi-backend/project/Build.scala:162]: I guess so. I don't know enough about Build.scala, difference between play.Project and sbt.Project, etc.

**Comment by andrewconner** [comment]
Feel free to merge. I didn't merge in case you wanted to monitor it when it's deployed.

**Pull request merged**